src/custom-element.js

```diff
@@ -637,13 +641,9 @@ function createBaseCustomElementClass(win) {
       this.assertLayout_();
       this.implementation_.layout_ = this.layout_;
       this.implementation_.layoutWidth_ = this.layoutWidth_;
-      if (this.everAttached) {
-        this.implementation_.firstAttachedCallback();
```

Comment:
I don't see that `completeUpgrade` is called on a node reinsertion. I.e. when `everAttached = true`